tools/slack_tools/collect_prs.py

```diff
@@ -0,0 +1,192 @@
+# Script for collecting PRs in need of review, and informing maintainers via
+# slack
+
+from __future__ import print_function
+
+import datetime
+import os
+import sys
+
+import github
+
+try:
+    input = raw_input  # Python 2
+except NameError:
+    pass  # Python 3
+
+maintainers = {
+    'alyssawilk': 'U78RP48V9',
+    'mattklein123': 'U5CALEVSL',
+    'lizan': 'U79E51EQ6',
+    'snowp': 'U93KTPQP6',
+    'ggreenway': 'U78MBV869',
+    'htuch': 'U78E7055Z',
+    'zuercher': 'U78J72Q82',
+    'phlax': 'U017PLM0GNQ',
+    'jmarantz': 'U80HPLBPG',
+    'antoniovicente': 'UKVNCQ212',
+    'junr03': 'U79K0Q431',
+    'wrowe': 'UBQR8NGBS',
+    'yanavlasov': 'UJHLR5KFS',
+    'asraa': 'UKZKCFRTP',
+}
+
+
+def get_slo_hours():
+    # on Monday, allow for 24h + 48h
+    if datetime.date.today().weekday() == 0:
+        return 72
+    return 24
+
+
+def track_prs():
+    git = github.Github()
+    repo = git.get_repo('envoyproxy/envoy')
+
+    # The list of PRs tagged with the waiting label
+    waiting_prs = []
+    # The list of PRs which are not waiting, but are well within review SLO
+    recent_prs = []
+    # A dict of maintainer : outstanding_pr_string to be sent to slack
+    maintainers_and_prs = {}
+    # A placeholder for unassigned PRs, to be sent to #maintainers eventually
+    maintainers_and_prs['unassigned'] = ""
+    # Out-SLO PRs to be sent to #envoy-maintainer-oncall
+    stalled_prs = ""
+
+    # Snag all PRs, including drafts
+    for pr_info in repo.get_pulls("open"):
+
+        pr_title = pr_info.title
+        pr_url = pr_info.html_url
+
+        # If the PR is waiting, collect it in that list and continue.
+        waiting = False
+        for label in pr_info.labels:
+            if label.name == 'waiting':
+                waiting_prs.append("PR %s' is tagged as waiting" % (pr_title))
+                waiting = True
+                continue
+        if (waiting):
+            continue
+
+        # If the PR hasn't been updated recently, don't nag.
+        # We can revisit this time, and always warn.
+        now = datetime.datetime.now()
```

Comment:
If folks are likely to use this as their daily reminder, maybe we should warn for anything assigned?  it is nice to have one place to look through

Reply:
Actually I'm just going to update this one now.

Thing I don't understand, how the datetime returned by github doesn't come with a time zone - I did not expect to have to do the shifting manually but the timestamps were off unless I did.  If someone with better python skills has ideas lmk